src/main/java/com/hermesworld/ais/galapagos/topics/service/impl/TopicServiceImpl.java

```diff
@@ -443,8 +443,12 @@ public CompletableFuture<SchemaMetadata> addTopicSchemaVersion(String environmen
                 }
 
                 boolean reverse = metadata.getType() == TopicType.COMMANDS;
-                SchemaUtil.verifyCompatibleTo(reverse ? newSchema : previousSchema,
-                        reverse ? previousSchema : newSchema);
+
+                if (!skipCompatCheck) {
```

Comment:
Could be included in condition in line 435 - whole block does not need to be executed if `skipCompatCheck` is active.